ClemBot.Api/ClemBot.Api.Core/Features/Users/Bot/Reminders.cs

```diff
@@ -0,0 +1,64 @@
+﻿using ClemBot.Api.Common;
+using ClemBot.Api.Data.Contexts;
+using FluentValidation;
+using Microsoft.EntityFrameworkCore;
+using NodaTime;
+
+namespace ClemBot.Api.Core.Features.Users.Bot;
+
+public class Reminders
+{
+    public class Validator : AbstractValidator<Query>
+    {
+        public Validator()
+        {
+            RuleFor(q => q.UserId).NotNull();
+        }
+    }
+
+    public class ReminderDto : IResponseModel
+    {
+        public int Id { get; set; }
+
+        public string Link { get; set; } = null!;
+
+        public string? Content { get; set; }
+
+        public LocalDateTime Time { get; set; }
+
+        public ulong UserId { get; set; }
+    }
+
+    public class Query : IRequest<IQueryResult<List<ReminderDto>>>
+    {
+        public ulong UserId { get; set; }
+    }
+
+    public class Handler : IRequestHandler<Query, IQueryResult<List<ReminderDto>>>
+    {
+
+        private readonly ClemBotContext _context;
+
+        public Handler(ClemBotContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<IQueryResult<List<ReminderDto>>> Handle(Query request, CancellationToken cancellationToken)
+        {
+            var reminders = await _context.Reminders
+                .Where(r => r.UserId == request.UserId && !r.Dispatched)
+                .Select(item => new ReminderDto
+                {
+                    Id = item.Id,
+                    Link = item.Link,
+                    Content = item.Content,
+                    Time = item.Time,
+                    UserId = item.UserId
+                })
+                .ToListAsync();
+
+            return QueryResult<List<ReminderDto>>.Success(reminders);
```

Comment:
we need a not found check here

Reply:
If there are no reminders for that user it will simply return an empty list - which the cog checks and handles.
```py
reminders = await self.bot.user_route.get_reminders(ctx.author.id)
if len(reminders) == 0:
    embed = discord.Embed(title='⏰ Reminder', color=Colors.ClemsonOrange,
                          description='You have no existing reminders.')
    embed.set_footer(text=str(ctx.author), icon_url=ctx.author.display_avatar.url)
    return await ctx.send(embed=embed)```

Reply:
I meant a not found for the user id, ToList will always return a list, we will get null if the user doesn't exist